addons/account/models/account_move.py

```diff
@@ -4951,6 +4951,15 @@ def _is_downpayment(self):
         '''
         return False
 
+    def _refunds_origin_required(self):
+        return False
+
+    def _link_invoices_origin_if_needed(self, switched_moves):
+        """ When it is important to mention the refunded invoice, we make it possible here. """
+        original_invoices = self.filtered(lambda m: m.move_type == 'out_invoice')
+        if len(original_invoices) == 1 and len(switched_moves) == 1 and original_invoices._refunds_origin_required():
```

Comment:
```suggestion
        if len(original_invoices) == 1 and len(switched_moves) == 1 and original_invoices._refunds_origin_required() and switched_moves.invoice_line_ids.sale_line_ids in original_invoices.invoice_line_ids.sale_line_ids:
```
I think we should add smth like this to make sure the lines of the credit note are a subset of the lines of the invoices